### PR TITLE
Series Admin User Interface

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#70879](https://central.tri.be/issues/70879)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/2af7c970e420683fed5ecbed15bc36b6e5bf07bf/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_singular_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/2af7c970e420683fed5ecbed15bc36b6e5bf07bf/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_plural_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/2af7c970e420683fed5ecbed15bc36b6e5bf07bf/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/2af7c970e420683fed5ecbed15bc36b6e5bf07bf/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_singular_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_plural_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_singular_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_plural_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/258597ed2c2f076686da3f354cef63f8ec31194d/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

File: src/admin-views/events-meta-box/date-time.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		<td colspan="2" class="tribe_sectionheader">
+			<div class="tribe_sectionheader" style="">
+				<h4><?php esc_html_e( 'Time &amp; Date', 'the-events-calendar' ); ?></h4></div>
+		</td>
+	</tr>
+	<tr>
+		<td colspan="2">
+			<table class="eventtable">
+				<tr id="recurrence-changed-row">
+					<td colspan='2'><?php printf( esc_html__( 'You have changed the recurrence rules of this %1$s.  Saving the %1$s will update all future %2$s.  If you did not mean to change all %2$s, then please refresh the page.', 'the-events-calendar' ), $events_label_singular_lowercase, $events_label_plural_lowercase ); ?></td>
```
Comment (sc0ttkclark) [inline src/admin-views/events-meta-box/date-time.php]: Using esc_html__ here only escapes the translation string, not the singular/plural labels. You ought to wrap those in their own esc_html(), or run `echo esc_html( sprintf( .... ) );` instead.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/f1cebcdb457aa7805e6537fe5d6e516de4cd1675/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_singular_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/f1cebcdb457aa7805e6537fe5d6e516de4cd1675/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$events_label_plural_lowercase' | [date-time.php:12](https://github.com/moderntribe/the-events-calendar/blob/f1cebcdb457aa7805e6537fe5d6e516de4cd1675/src/admin-views/events-meta-box/date-time.php#L12)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/f1cebcdb457aa7805e6537fe5d6e516de4cd1675/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/9a7568c0c606085fb6175a47217a6cbcb750f742/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/9a7568c0c606085fb6175a47217a6cbcb750f742/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
Ready for another 👁 👁

File: src/Tribe/Admin/Event_Meta_Box.php [hunk 2]
```diff
@@ -325,13 +327,64 @@ protected function do_meta_box() {
...
+	 * parts.
+	 *
+	 * @param $template_path
+	 */
+	protected function do_template_part( $template_path ) {
+		if ( ! file_exists( $template_path ) ) {
+			return;
+		}
+
+		extract( $this->vars );
```
Comment (borkweb) [inline src/Tribe/Admin/Event_Meta_Box.php]: Let's change this to `extract( $this->vars, EXTR_SKIP );`

Comment (borkweb) [inline src/Tribe/Admin/Event_Meta_Box.php]: That way, `$template_path` won't be overridable.

**Comment by bordoni** [comment]
Ready for another 👁 👁

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:383](https://github.com/moderntribe/the-events-calendar/blob/3be6f5a499c757dfb0e2e7ae6880c0fd3cf99a31/src/Tribe/Admin/Event_Meta_Box.php#L383)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/3be6f5a499c757dfb0e2e7ae6880c0fd3cf99a31/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Admin/Event_Meta_Box.php [hunk 3]
```diff
@@ -325,13 +327,64 @@ protected function do_meta_box() {
...
+		$this->do_template_part( $events_meta_box_template );
+	}
+
+	/**
+	 * Injects the date/time settings section into the events meta box.
+	 *
+	 * This normally only happens when the meta box renders when an actual event is being
+	 * edited (as opposed to other related post types).
+	 *
+	 * @param int $post_id
```
Comment (barryhughes) [inline src/Tribe/Admin/Event_Meta_Box.php]: `@since`

File: src/Tribe/Admin/Event_Meta_Box.php [hunk 4]
```diff
@@ -325,13 +327,64 @@ protected function do_meta_box() {
...
+	public function date_time_section( $post_id ) {
+		$should_render = (
+			Tribe__Events__Main::POSTTYPE === get_post_type( $post_id )
+			|| Tribe__Events__Main::POSTTYPE === $GLOBALS['typenow']
+		);
+
+		/**
+		 * Controls whether the date time section is injected into the events meta box or not.
+		 *
+		 * @var bool $should_render
```
Comment (barryhughes) [inline src/Tribe/Admin/Event_Meta_Box.php]: `@since`

Comment (barryhughes) [inline src/Tribe/Admin/Event_Meta_Box.php]: should be `@param` I think, not `@var`

File: src/Tribe/Admin/Event_Meta_Box.php [hunk 5]
```diff
@@ -325,13 +327,64 @@ protected function do_meta_box() {
...
+		 * @var int  $event_id
+		 */
+		if ( ! apply_filters( 'tribe_events_event_meta_box_render_date_time_section', $should_render, $post_id ) ) {
+			return;
+		}
+
+		/**
+		 * Controls the path to the date time section template (a component of the events meta box).
+		 *
+		 * @var string $template_path
```
Comment (barryhughes) [inline src/Tribe/Admin/Event_Meta_Box.php]: Ditto re `@param` and `@since`

File: src/admin-views/events-meta-box/date-time.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+						<input
+							autocomplete="off"
+							tabindex="<?php tribe_events_tab_index(); ?>"
+							type="text"
+							class="tribe-timepicker tribe-field-start_time"
+							name="EventStartTime"
+							id="EventStartTime"
+						<?php echo Tribe__View_Helpers::is_24hr_format() ? 'data-format="H:i"' : '' ?>"
+						data-step="<?php echo esc_attr( $start_timepicker_step ); ?>"
```
Comment (barryhughes) [inline src/admin-views/events-meta-box/date-time.php]: Indent seems funky around here: shouldn't these be tabbed in to match the other attributes for the same element?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:395](https://github.com/moderntribe/the-events-calendar/blob/59b4ff5ddc08b77a7f9111f033a5b0681bc1f16b/src/Tribe/Admin/Event_Meta_Box.php#L395)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/59b4ff5ddc08b77a7f9111f033a5b0681bc1f16b/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

File: src/admin-views/events-meta-box/date-time.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+						<span class="helper-text hide-if-js"><?php esc_html_e( 'YYYY-MM-DD', 'the-events-calendar' ) ?></span>
+
+						<input
+							autocomplete="off"
+							tabindex="<?php tribe_events_tab_index(); ?>"
+							type="text"
+							class="tribe-timepicker tribe-field-start_time"
+							name="EventStartTime"
+							id="EventStartTime"
+						<?php echo Tribe__View_Helpers::is_24hr_format() ? 'data-format="H:i"' : '' ?>"
```
Comment (barryhughes) [inline src/admin-views/events-meta-box/date-time.php]: Actually, isn't there also a stray `"` at the end of the line, besides the indent being out?

**Comment by bordoni** [comment]
Should be ready for another look

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Extract is the devil. Use separate assignments or access the array directly instead. | [Event_Meta_Box.php:395](https://github.com/moderntribe/the-events-calendar/blob/4bb2ca836931f3ac1cd982705c92c19d5623cee6/src/Tribe/Admin/Event_Meta_Box.php#L395)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [date-time.php:75](https://github.com/moderntribe/the-events-calendar/blob/4bb2ca836931f3ac1cd982705c92c19d5623cee6/src/admin-views/events-meta-box/date-time.php#L75)

> via jenkins/codesniffer-phpcs

**Pull request merged**